src/internal/types.ts

```diff
@@ -10,15 +11,66 @@ export type FactoryOrValue<T> = T | (() => T);
 
 export interface MonoTypeOperatorFunction<T> extends OperatorFunction<T, T> {}
 
+/** SUBSCRIPTION INTERFACES */
+
+export interface Unsubscribable {
+  unsubscribe(): void;
+}
+
+export type TeardownLogic = Unsubscribable | Function | void;
+
+export interface SubscriptionLike extends Unsubscribable {
+  unsubscribe(): void;
+  readonly closed: boolean;
+}
+
 export interface Subscribable<T> {
   subscribe(observerOrNext?: PartialObserver<T> | ((value: T) => void),
             error?: (error: any) => void,
-            complete?: () => void): AnonymousSubscription;
+            complete?: () => void): Unsubscribable;
 }
 
-export type ObservableLike<T> = { [Symbol.observable]: () => Subscribable<T>; };
 export type SubscribableOrPromise<T> = Subscribable<T> | Subscribable<never> | PromiseLike<T> | ObservableLike<T>;
+
+/** OBSERVABLE INTERFACES */
+
+export interface Subscribable<T> {
+  subscribe(observerOrNext?: PartialObserver<T> | ((value: T) => void),
+            error?: (error: any) => void,
+            complete?: () => void): Unsubscribable;
+}
+
 export type ObservableInput<T> = SubscribableOrPromise<T> | ArrayLike<T> | Iterable<T>;
+export type ObservableLike<T> = { [Symbol.observable]: () => Subscribable<T>; };
+
+/** OBSERVER INTERFACES */
+
+export interface NextObserver<T> {
+  closed?: boolean;
```

Comment:
@benlesh @kwonoj @mattpodwysocki should `closed` be on the public interface?

Reply:
I believe this is closest to the `SubscriptionObserver` interface defined here:

https://github.com/tc39/proposal-observable#subscriptionobserver

Based on how `PartialObserver` is used, I think exposing `closed` here is okay.